### PR TITLE
[owncloud] Make Nextcloud 17.0 the default release

### LINKED ISSUE #1192: [debops.owncloud] Fails to set HTTP header X-Frame-Options at SAMEORIGIN
Get to fresh install nextcloud 16 instance with debops.ansible-owncloud , got warning 
>     The "X-Frame-Options" HTTP header is not set to "SAMEORIGIN". This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.

After looking at the code, I can't get the point with omitting this variable when version owncloud > 12 ?
l. 68 of  debops/ansible/roles/debops.owncloud/defaults/main.yml 
>    frame_options: '{{ omit if (owncloud__variant in ["nextcloud"] and owncloud__release is version_compare("12.0", ">=")) else "SAMEORIGIN" }}'"

Couldn't be simple to set SAMEORIGIN all the time ?

### PR DESCRIPTION
**Pull request by imrejonk**
These changes add support for Nextcloud 17.0 and 18.0 and make
Nextcloud 17.0 the default release for new installations. The previous
major release, Nextcloud 16.0, will become end-of-life next month.

There are no major changes required when updating from Nextcloud 16.0.
The NGINX configuration has been updated to reflect the setup as
recommended by the Nextcloud community [1].

[1] https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html#nextcloud-in-the-webroot-of-nginx

Closes #1192

### CODE REVIEW

**Comment by bleuchtang** [comment]
This PR will close debops/debops/issues/1192

**Review by drybjed — COMMENTED** [review]
Changes look fine to me, nice work! Waiting for @ypid's review before merging.

File: ansible/roles/owncloud/defaults/main.yml
Comment (SuperSandro2000) [inline ansible/roles/owncloud/defaults/main.yml:2155]: Isn't latest 18 and not 17?

Comment (imrejonk) [reply ansible/roles/owncloud/defaults/main.yml:2155]: Nextcloud 18.0 is indeed the latest release, but I've had more time testing the role with Nextcloud 17.0, which is what I now use for my own systems. Nextcloud 17.0 is currently in the 'production' update channel. This means that it is considered suitable for enterprise use as no potentially breaking changes will be introduced anymore, which in turn makes it a favourable choice for system administrators who happen to choose Debian for its stability presumption. Nextcloud 17.0 is therefore a safe default for this role.

See also [DEP 3](https://docs.debops.org/en/master/dep/dep-0003.html).

The Nextcloud 19.0 release is scheduled for May 21st. We can do some work around that time to make Nextcloud 18.0 the default for this role, and start preparing things for the next major release.

Comment (SuperSandro2000) [reply ansible/roles/owncloud/defaults/main.yml:2155]: My intention was to update the documentation link to ``17` from ``latest``.

File: ansible/roles/owncloud/defaults/main.yml
Comment (SuperSandro2000) [inline ansible/roles/owncloud/defaults/main.yml]:
```suggestion
  ## https://docs.nextcloud.com/server/17/admin_manual/installation/nginx.html
```

File: ansible/roles/owncloud/defaults/main.yml
Comment (imrejonk) [reply ansible/roles/owncloud/defaults/main.yml:2155]: Ah, check. I just pushed a commit to update the links to upstream documentation (version 17 for Nextcloud, 10.3 for ownCloud).

**Comment by ypid** [comment]
> Waiting for @ypid's review before merging.

I had a peek. Thanks for your work! I will do some hands-on testing the next days.

**Review by ypid — CHANGES_REQUESTED** [review]
Looks good so far. I reviewed the changes done to the NC docs and included the remaining changes to the role. Please have a look at https://github.com/imrejonk/debops/pull/1 and merge it if you agree. Once merged, it should appear here :)

**Comment by imrejonk** [comment]
@ypid ah, you found a more recent version of the nginx community documentation. Looks like they changed a location regex match to include mp4 and webm files. Not sure if this change is already necessary for Nextcloud 17, but it can't harm to include it.

Merged!

**Review by ypid — APPROVED** [review]
LGTM. I updated my NC instance. No issues noticed.

**Review by drybjed — APPROVED** [review]
I tested the changes on an existing installation, everything seems to work fine. :+1:

**Pull request merged**
